packages/design-system/src/styles/components/_Button.scss

```diff
@@ -13,49 +13,55 @@
   font-weight: var(--fontWeight, $button__font-weight);
   line-height: 1.3;
   text-align: center;
-  text-decoration-color: transparent;
   display: inline-block;
   padding: var(--space, $spacer-1 $spacer-3);
 
+  &:is(a) {
+    text-decoration: none;
+
+    &:focus {
+      background-color: var(--backgroundColor, $button__background-color);
+      text-decoration: none;
+    }
+
+    &[aria-disabled] {
+      pointer-events: none; // Needed to render disabled links correctly
+    }
+  }
+
   > svg {
     @include inline-icon;
   }
 
   &:hover,
-  &.ds-c-button--hover,
-  &:hover:focus,
-  &.ds-c-button--hover.ds-c-button--focus {
+  &:hover:focus {
     background-color: var(--backgroundColor--h, $button__background-color--hover);
     border-color: var(--borderColor--h, $button__border-color--hover);
     color: var(--color--h, $button__color--hover);
   }
 
-  &:focus,
-  &.ds-c-button--focus {
+  &:focus {
     @include focus-styles;
   }
 
   &:active,
-  &.ds-c-button--active,
-  &:active:focus,
-  &.ds-c-button--active:focus {
+  &:active:focus {
     background-color: var(--backgroundColor--a, $button__background-color--active);
     border-color: var(--borderColor--a, $button__border-color--active);
     color: var(--color--a, $button__color--active);
   }
 
   // Make sure active overrides focus styles
-  &:active:focus,
-  &.ds-c-button--active:focus {
+  &:active:focus {
     box-shadow: none;
   }
 
   &:disabled,
-  &.ds-c-button--disabled {
+  &[aria-disabled] {
```

Comment:
I like the idea of tying visual state changes to aria attrs as it gives devs a better visual queue when a11y isn't working quite right.